In [2]:
import pandas as pd

In [6]:
survey = pd.read_csv(r"F:\python numpy project\Survey Dataset - Technical Interview.csv")

In [7]:
survey.head(4)

response_id collection_date urban_rural  education               income  \
0            1       5-24-2017       Rural  12th Pass  ₹ 30,000 - ₹ 50,000   
1            2       5-24-2017       Rural   Graduate  ₹ 20,000 - ₹ 30,000   
2            3       5-24-2017       Rural   Graduate   ₹ 5,000 - ₹ 10,000   
3            4       5-24-2017       Urban   Graduate  ₹ 10,000 - ₹ 20,000   

  Vote_Now Past_Vote     CM_satisfaction    MLA_satisfaction age gender  \
0      BJP       BJP  Fully Dissatisfied  Fully Dissatisfied  26   Male   
1     RLSP     JD(U)  Somewhat Satisfied  Fully Dissatisfied  34   Male   
2   JAP(L)       BJP  Fully Dissatisfied  Somewhat Satisfied  25   Male   
3      RJD       RJD  Fully Dissatisfied  Somewhat Satisfied  36   Male   

   assembly_no category    weight  
0          211      Gen  0.346969  
1          189       BC  0.603419  
2          191      Gen  0.673762  
3          194      Gen  1.101298

In [6]:
survey["collection_date"]=pd.to_datetime(survey['collection_date'])

In [13]:
survey["age"].unique()

array(['26', '34', '25', '36', '22', '38', '23', '42', '51', '29', '24',
       '30', '39', '37', '19', '44', '53', '32', '21', '18', '20', '27',
       '48', '28', '45', '74', '35', '31', '70', '40', '49', '46', '43',
       '41', '33', '58', '24ko', '56', '50', '55', '54', '62', '60', '59',
       '47', '61', '52', '66', '57', '67', '65', '71', '63', '64', '68',
       '69', '99', '72', '75', '76'], dtype=object)

In [14]:
survey["age"].replace(to_replace='24ko',value=24,inplace=True)

In [15]:
survey["age"]=survey["age"].astype(int)

In [16]:
survey.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6867 entries, 0 to 6866
Data columns (total 14 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   response_id       6867 non-null   int64         
 1   collection_date   6867 non-null   datetime64[ns]
 2   urban_rural       6867 non-null   object        
 3   education         6867 non-null   object        
 4   income            6867 non-null   object        
 5   Vote_Now          6867 non-null   object        
 6   Past_Vote         6867 non-null   object        
 7   CM_satisfaction   6867 non-null   object        
 8   MLA_satisfaction  6867 non-null   object        
 9   age               6867 non-null   int32         
 10  gender            6867 non-null   object        
 11  assembly_no       6867 non-null   int64         
 12  category          6867 non-null   object        
 13  weight            6867 non-null   float64       
dtypes: datetime64[ns](1), fl

In [18]:
survey.head(2)

response_id collection_date urban_rural  education               income  \
0            1      2017-05-24       Rural  12th Pass  ₹ 30,000 - ₹ 50,000   
1            2      2017-05-24       Rural   Graduate  ₹ 20,000 - ₹ 30,000   

  Vote_Now Past_Vote     CM_satisfaction    MLA_satisfaction  age gender  \
0      BJP       BJP  Fully Dissatisfied  Fully Dissatisfied   26   Male   
1     RLSP     JD(U)  Somewhat Satisfied  Fully Dissatisfied   34   Male   

   assembly_no category    weight  
0          211      Gen  0.346969  
1          189       BC  0.603419

In [75]:
survey["urban_rural"].value_counts()

Rural    74.777923
Urban    25.222077
Name: urban_rural, dtype: float64

In [8]:
survey[(survey.urban_rural == "Rural")  &  (survey.gender == "Male")]

response_id collection_date urban_rural               education  \
0               1       5-24-2017       Rural               12th Pass   
1               2       5-24-2017       Rural                Graduate   
2               3       5-24-2017       Rural                Graduate   
4               5       5-24-2017       Rural                Graduate   
7               8       5-24-2017       Rural  Professional Education   
...           ...             ...         ...                     ...   
6861         6862        6-1-2017       Rural                Graduate   
6862         6863        6-1-2017       Rural               10th Pass   
6863         6864        6-1-2017       Rural                Graduate   
6865         6866        6-1-2017       Rural  Professional Education   
6866         6867        6-1-2017       Rural               10th Pass   

                   income Vote_Now     Past_Vote        CM_satisfaction  \
0     ₹ 30,000 - ₹ 50,000      BJP           BJP     Fully Dissatisfied   
1     ₹ 20,000 - ₹ 30,000     RLSP         JD(U)     Somewhat Satisfied   
2      ₹ 5,000 - ₹ 10,000   JAP(L)           BJP     Fully Dissatisfied   
4     ₹ 50,000 - ₹ 80,000    JD(U)  Did not vote        Fully Satisfied   
7      ₹ 1,50,000 & above      BJP         JD(U)  Somewhat Dissatisfied   
...                   ...      ...           ...                    ...   
6861    BPL/Below ₹ 5,000      BJP           BJP     Fully Dissatisfied   
6862    BPL/Below ₹ 5,000    JD(U)           BJP        Fully Satisfied   
6863    BPL/Below ₹ 5,000      RJD           RJD     Fully Dissatisfied   
6865  ₹ 10,000 - ₹ 20,000      BJP  Did not vote        Fully Satisfied   
6866   ₹ 5,000 - ₹ 10,000   JAP(L)        JAP(L)        Fully Satisfied   

           MLA_satisfaction age gender  assembly_no category    weight  
0        Fully Dissatisfied  26   Male          211      Gen  0.346969  
1        Fully Dissatisfied  34   Male          189       BC  0.603419  
2        Somewhat Satisfied  25   Male          191      Gen  0.673762  
4           Fully Satisfied  22   Male          176       BC  0.333505  
7     Somewhat Dissatisfied  34   Male          186       ST  0.340652  
...                     ...  ..    ...          ...      ...       ...  
6861     Somewhat Satisfied  22   Male          166      Gen  0.080137  
6862     Fully Dissatisfied  25   Male          166      Gen  0.036329  
6863     Fully Dissatisfied  30   Male          169       BC  0.252780  
6865     Somewhat Satisfied  23   Male          243      Gen  0.080137  
6866        Fully Satisfied  33   Male          241      Gen  1.797518  

[5032 rows x 14 columns]

In [23]:
survey.isnull().sum()

response_id         0
collection_date     0
urban_rural         0
education           0
income              0
Vote_Now            0
Past_Vote           0
CM_satisfaction     0
MLA_satisfaction    0
age                 0
gender              0
assembly_no         0
category            0
weight              0
dtype: int64

In [24]:
survey.dropna(how="all")

response_id collection_date urban_rural               education  \
0               1      2017-05-24       Rural               12th Pass   
1               2      2017-05-24       Rural                Graduate   
2               3      2017-05-24       Rural                Graduate   
3               4      2017-05-24       Urban                Graduate   
4               5      2017-05-24       Rural                Graduate   
...           ...             ...         ...                     ...   
6862         6863      2017-06-01       Rural               10th Pass   
6863         6864      2017-06-01       Rural                Graduate   
6864         6865      2017-06-01       Urban                Graduate   
6865         6866      2017-06-01       Rural  Professional Education   
6866         6867      2017-06-01       Rural               10th Pass   

                   income Vote_Now     Past_Vote     CM_satisfaction  \
0     ₹ 30,000 - ₹ 50,000      BJP           BJP  Fully Dissatisfied   
1     ₹ 20,000 - ₹ 30,000     RLSP         JD(U)  Somewhat Satisfied   
2      ₹ 5,000 - ₹ 10,000   JAP(L)           BJP  Fully Dissatisfied   
3     ₹ 10,000 - ₹ 20,000      RJD           RJD  Fully Dissatisfied   
4     ₹ 50,000 - ₹ 80,000    JD(U)  Did not vote     Fully Satisfied   
...                   ...      ...           ...                 ...   
6862    BPL/Below ₹ 5,000    JD(U)           BJP     Fully Satisfied   
6863    BPL/Below ₹ 5,000      RJD           RJD  Fully Dissatisfied   
6864   ₹ 5,000 - ₹ 10,000   JAP(L)           RJD           Can’t say   
6865  ₹ 10,000 - ₹ 20,000      BJP  Did not vote     Fully Satisfied   
6866   ₹ 5,000 - ₹ 10,000   JAP(L)        JAP(L)     Fully Satisfied   

        MLA_satisfaction  age gender  assembly_no category    weight  
0     Fully Dissatisfied   26   Male          211      Gen  0.346969  
1     Fully Dissatisfied   34   Male          189       BC  0.603419  
2     Somewhat Satisfied   25   Male          191      Gen  0.673762  
3     Somewhat Satisfied   36   Male          194      Gen  1.101298  
4        Fully Satisfied   22   Male          176       BC  0.333505  
...                  ...  ...    ...          ...      ...       ...  
6862  Fully Dissatisfied   25   Male          166      Gen  0.036329  
6863  Fully Dissatisfied   30   Male          169       BC  0.252780  
6864  Fully Dissatisfied   20   Male          241      Gen  1.812926  
6865  Somewhat Satisfied   23   Male          243      Gen  0.080137  
6866     Fully Satisfied   33   Male          241      Gen  1.797518  

[6867 rows x 14 columns]

In [44]:
survey[survey["gender"]=='Male'].count()/survey.count() * 100

response_id         97.932139
collection_date     97.932139
urban_rural         97.932139
education           97.932139
income              97.932139
Vote_Now            97.932139
Past_Vote           97.932139
CM_satisfaction     97.932139
MLA_satisfaction    97.932139
age                 97.932139
gender              97.932139
assembly_no         97.932139
category            97.932139
weight              97.932139
dtype: float64

In [45]:
male_count = survey["gender"].value_counts()["Male"]

In [48]:
total_count = survey["gender"].count()

In [50]:
male_percentage = (male_count / total_count) * 100

In [51]:
male_percentage

97.9321392165429

In [55]:
survey["Vote_Now"].value_counts()/survey["Vote_Now"].count() * 100

BJP           34.469201
RJD           29.066550
JD(U)          7.659822
NOTA           6.669579
JAP(L)         4.558031
Plurals        3.160041
INC            3.145478
Undecided      2.388234
RJJP           2.067861
LJP            1.485365
Others         1.266929
RLSP           1.194117
Left Front     1.150430
AIMIM          0.538809
VIP            0.407747
AAP            0.393185
BSP            0.218436
HAM            0.160186
Name: Vote_Now, dtype: float64

In [57]:
survey["education"].value_counts()/6867 * 100

Graduate                  48.376292
12th Pass                 19.367992
Post - Graduation         17.635066
Professional Education     5.883210
10th Pass                  5.795835
8th Pass                   2.606670
Illiterate                 0.334935
Name: education, dtype: float64

In [69]:
survey["CM_satisfaction"].value_counts()/6867  * 100

Fully Dissatisfied       43.861948
Somewhat Satisfied       32.299403
Fully Satisfied          10.499490
Somewhat Dissatisfied     9.902432
Can’t say                 3.436726
Name: CM_satisfaction, dtype: float64

In [70]:
survey["MLA_satisfaction"].value_counts()/6867  * 100

Fully Dissatisfied       60.463084
Somewhat Satisfied       17.008883
Fully Satisfied           9.756808
Somewhat Dissatisfied     7.674385
Can’t say                 5.096840
Name: MLA_satisfaction, dtype: float64

In [73]:
survey["category"].value_counts() / 6867  * 100

Gen    44.429882
BC     34.367264
EBC    15.873016
SC      4.456094
ST      0.873744
Name: category, dtype: float64

In [ ]:
top 3 category for highest voting is gen,bc,ebc in 44%,34%,15% 

# 

# 

# The key takeaways from the data :

"The analysis of the dataset reveals that 75% of the voting in the survey comes from the rural areas,
while the remaining 25% of the votes come from urban areas."

"The analysis of the dataset shows that male respondents were the 
highest contributors in the voting process with a percentage of 98%."

"The analysis of the dataset reveals that the highest number of votes in the current data have been received by
BJP, RJD, and JD(U) with approximately 35%, 29%, and 7% respectively."

"The analysis of the dataset indicates that the top three education categories for the highest number of voters are
Graduate, 12th Pass, and Post-Graduation in that order, 
representing approximately 48%, 19%, and 17% of the total voters respectively."

"The analysis of the dataset shows that 44% of the people surveyed were fully dissatisfied with the Chief Minister's
satisfaction, while 60% of the people were fully dissatisfied with the MLA's satisfaction."

"The analysis of the dataset indicates that the top three categories with the highest number of votes are General (GEN), Backward 
Class (BC), and Extremely Backward Class (EBC), representing approximately 44%, 34%, and 15% of the total voters respectively."

# Final Conclution

Based on the analysis of the dataset, it can be concluded that BJP, RJD, and JD(U) are the top three parties that received 
the highest number of votes.
The majority of the voters are males, and the top three categories with the highest number of votes
are General (GEN), Backward Class (BC), and Extremely Backward Class (EBC). 
The majority of the votes (75%) come from ruralareas, and the top three education levels of the voters are Graduate, 12th Pass, and Post-Graduation. 
The dataset also reveals that a significant percentage of the voters are dissatisfied with the Chief Minister's and MLA's satisfaction.